examples/ZeroUserCodeLightup/Dockerfile

```diff
@@ -0,0 +1,24 @@
+﻿FROM microsoft/aspnetcore-build:2.0.5-2.1.4 AS build-env
+WORKDIR /app
+
+# Copy csproj and restore as distinct layers
+COPY *.csproj ./
+
+# Adding a reference to hosting startup package
+RUN dotnet add package Microsoft.ApplicationInsights.Kubernetes.HostingStartup
+RUN dotnet restore
+
+# Copy everything else and build
+COPY . ./
```

Comment:
@karolz-ms That is an interesting question. To be honest, I got the script template from [Dockerize a .NET Core application](https://docs.docker.com/engine/examples/dotnetcore/#prerequisites).
Now that I think about it, I feel the reason behind might be failing faster: if the project doesn't exist, we save time copying files.
What do you think the reason is?

Reply:
I am not sure, but I do not think the current double-copy technique adds much value

Reply:
As a matter of fact, I need to make it the same step because otherwise, the copy of all the files overwrites the project file that modified by `dotnet add package`.